index.js

```diff
@@ -231,4 +231,55 @@ transform.resetTypes = function resetTypes() {
   registerTypes = {};
 };
 
-module.exports = transform;
+function addPermissions(permissions, key, type) {
+  return permissions[key] && permissions[key][type] !== undefined ? permissions[key][type] : permissions[type]
+}
+
+function getFormOptions(properties, fieldsData = {}, permissions, newOptions = {}, first = false) {
+  Object.entries(properties).forEach((data) => {
+      const [key, value] = data
+      if (first) {
+          newOptions[key] = {
+              fields: [],
+              order: fieldsData[key] ? fieldsData[key].order : []
+          }
+      } else {
+          // If field doesn't exist, create it
+          if (!fieldsData[key]) {
+              fieldsData[key] = {}
+          }
+          if (!fieldsData[key]["ui:component"] || fieldsData[key]["ui:component"].includes('auto-fill')) {
+              fieldsData[key]["ui:component"] = 'string'
```

Comment:
We should  never assign `ui:component` in this transform, only read them. Is this from the old textbox thing that can be removed now?